### PR TITLE
Add an ElementJacobianDamper that runs on the undisplaced mesh

### LINKED ISSUE #21397: Element jacobian damper that runs on undisplaced mesh
## Reason

We all love element Jacobian damper. It has proven to be helpful for many mechanics simulations. However, the current element jacobian damper requires the existence of the displaced mesh. 

In total Lagrangian formulations, we don't need to create a displaced mesh (we can turn that off by not providing the `displacements` parameter in the Mesh block and GlobalParams), but we still want to use the element jacobian damper!

## Design

Intuitively, we should have enough information to compute the element jacobian change over nonlinear iterations. We just need to compute displacement gradients before and after the solution update, and then compute the change in the determinant of the deformation gradient. That's it.

## Impact

This will allow us to use element jacobian damper without a displaced mesh. This should save us a lot of time in `updating displaced mesh.................................` when the mesh is huge.

### PR DESCRIPTION
**Pull request by hugary1995**
close #21397

### CODE REVIEW

**Comment by hugary1995** [comment]
I am totally open to renaming this object.

File: modules/tensor_mechanics/src/dampers/ReferenceElementJacobianDamper.C [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    Coupleable(this, /*nodal=*/true),
+    _max_jacobian_diff(getParam<Real>("max_increment")),
+    _tid(getParam<THREAD_ID>("_tid")),
+    _mesh(_subproblem.mesh()),
+    _assembly(_subproblem.assembly(_tid)),
+    _qrule(_assembly.qRule()),
+    _ndisp(coupledComponents("displacements")),
+    _disp_num(coupledIndices("displacements"))
+{
+  _grad_phi.resize(_ndisp);
```
Comment (dschwen) [inline modules/tensor_mechanics/src/dampers/ReferenceElementJacobianDamper.C:42]:
```suggestion
  if (getParam<bool>("use_displaced_mesh"))
    paramError("use_displaced_mesh",
               "ReferenceElementJacobianDamper needs to act on an undisplaced mesh.");
  _grad_phi.resize(_ndisp);
```

Comment (hugary1995) [reply modules/tensor_mechanics/src/dampers/ReferenceElementJacobianDamper.C:42]: Ummm, no. `GeneralDamper` doesn't have the `use_displaced_mesh` option. Users won't be able to set that. If they set it they get an unused param warning.

Comment (dschwen) [reply modules/tensor_mechanics/src/dampers/ReferenceElementJacobianDamper.C:42]: Oh, yeah, makes sense!

**Review by dschwen — APPROVED** [review]
👍

**Pull request merged**